### ISSUE TITLE
Java 9: make sure Drombler FX works with Java SE 9

### ISSUE THREAD

**Issue by puce77** [Drombler/drombler-fx#172]
Make sure Drombler FX works with Java SE 9.

**Comment by puce77** [comment]
Blocked by: https://issues.apache.org/jira/browse/FELIX-5575

**Comment by claudio-rosati** [comment]
Will Java 9 modules system supersede OSGI?

**Comment by puce77** [comment]
AFAIK, Java 9 module system does not support to load different versions of the same class out-of-the-box (see "dependency hell" issue). And the module layer is only one of the three layer OSGi provides. That said, I'm looking also for easier alternatives, e.g. Java 9 module systems + CDI + some clever "plugin" classloader tricks?

For now we just have to make sure Drombler FX can combine OSGi with Java SE 9.

**Comment by claudio-rosati** [comment]
Thank you for the answer.
I'll look forward for the results of your investigation on alternatives.

**Comment by puce77** [comment]
OSGi R7 seems to have support for Java 9: https://blog.osgi.org/2018/02/osgi-r7-highlights-java-9-support.html

**Comment by puce77** [comment]
The upcoming Apache Felix v6.0.0 seems to support OSGi R7: https://issues.apache.org/jira/browse/FELIX-5791

**Comment by puce77** [comment]
Use Java SE 11 instead of Java SE 9

**Comment by maggu2810** [comment]
Perhaps that slides could be interesting: https://tomsondev.files.wordpress.com/2018/12/javafx-days.pdf

**Comment by puce77** [comment]
@maggu2810 : Thanks, I will have a look at it

**Comment by puce77** [comment]
Also see: https://github.com/osgi/design/blob/master/rfps/rfp-0196-OSGiConnect.pdf
-> new issue?

**Comment by puce77** [comment]
Blocked by https://issues.apache.org/jira/browse/FELIX-6287
It seems this has to wait for a OSGi Connect release.
